### ISSUE TITLE
Suggestion: Add other interaction types as seperate folders

### ISSUE THREAD

**Issue by cesarheuvelmans** [underctrl-io/commandkit#47]
Would love to see seperate handler paths for buttons, selectmenus, modalsubmits, contextmenus and autocompletes. With the filenames corresponding to the custom ids off those components. e.g.

**Comment by m1-dev** [comment]
Could you clarify more? Do you mean something like this?:
```
src/
    buttons/
    autocomplete/
    modals/
    menus/
        select/
        context/
```

**Comment by cesarheuvelmans** [comment]
Yes, exactly. And then in those folderes you would be able to place the handlers for those types of interactions. Basically the exact same way the commands and events folders work in commandKit now.

**Comment by m1-dev** [comment]
Personally I think this would turn CommandKit into some Next.js framework for discord.js, which doesn't make it beginner-friendly as it's currently advertised.

**Comment by twlite** [comment]
## On Project Structure

We plan to introduce a well-organized project setup that is easy to use over a long period. This is currently under internal discussion to find the best solution.

> I feel this might make CommandKit similar to the Next.js framework for discord.js, which may not be as beginner-friendly as it is now.

This idea isn't complex, considering CommandKit's design to automatically handle commands. However, having many `customIdExampleX.js` files can be confusing. We won't enforce this method, but we'll definitely provide an alternative.
